src/main/java/com/synopsys/arc/jenkinsci/plugins/extratoolinstallers/installers/RarExtractionInstaller.java

```diff
@@ -0,0 +1,157 @@
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package com.synopsys.arc.jenkinsci.plugins.extratoolinstallers.installers;
+
+import hudson.Extension;
+import hudson.FilePath;
+import hudson.ProxyConfiguration;
+import hudson.Functions;
+import hudson.model.Node;
+import hudson.model.TaskListener;
+import hudson.remoting.VirtualChannel;
+import hudson.tools.ToolInstallation;
+import hudson.tools.ToolInstallerDescriptor;
+import hudson.util.FormValidation;
+import jenkins.MasterToSlaveFileCallable;
+import java.io.File;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLConnection;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.QueryParameter;
+
+/**
+ * Installs a tool by downloading and unpacking a RAR file. 
+ * The installer is inspired by "ZipExtractionInstaller" from the Custom Tools plugin. 
+ * Limitation: Extracts only a single RAR file. A tool archive split into multiple RAR volumes will be not fully extracted.
+ *
+ * @author Martin Hjelmqvist <martin@hjelmqvist.eu>.
+ */
+public class RarExtractionInstaller extends AbstractExtraToolInstaller {
+
+    /**
+     * Messages displayed when configuring the tool in Jenkins. 
+     * TODO: Move this to Messages.java generated by the localizer.
+     */
+    private static final String RAR_EXTRACTION_INSTALLER_DISPLAY_NAME = "Extract *.rar";
+    private static final String RAR_EXTRACTION_INSTALLER_BAD_CONNECTION = "Server rejected connection.";
+    private static final String RAR_EXTRACTION_INSTALLER_MALFORMED_URL = "Malformed URL.";
+    private static final String RAR_EXTRACTION_INSTALLER_COULD_NOT_CONNECT = "Could not connect to URL.";
+    private static final String RAR_EXTRACTION_INSTALLER_NO_TOOLHOME = "Must provide a tool home directory.";
+
+    /**
+     * Constructs a RarExtractionInstaller.
+     *
+     * @param label Optional label for this installer.
+     * @param toolHome Path to the location of the tool archive.
+     * @param failOnSubstitution Fail the build if substitution fails.
+     */
+    @DataBoundConstructor
+    public RarExtractionInstaller(String label, String toolHome,
+            boolean failOnSubstitution) {
+        super(label, toolHome, failOnSubstitution);
+    }
+
+    /**
+     * Performs installation of tool on the node. 
+     * If the tool archive is e.g. corrupted, exception will be thrown from the JUnrar library.
```

Comment:
Which exception. Is there a risk of RunTime exceptions? If yes, maybe makes sense to wrap the call by <code>catch (RuntimeException)</code>